### PR TITLE
fix translation line wrapping bug

### PR DESCRIPTION
**Pull request by kelvinabrokwa**
re: https://github.com/mapbox/mapbox-gl-js/issues/1847

### CODE REVIEW

**Comment by kelvinabrokwa** [comment]
@ansis so when the x translation is >18 there is this crazy smear effect. I have no idea why this is happens...

image

File: js/symbol/shaping.js [hunk 1]
```diff
@@ -146,8 +146,8 @@ function justifyLine(positionedGlyphs, glyphs, start, end, justify) {
 
 }
 
-function align(positionedGlyphs, justify, horizontalAlign, verticalAlign, maxLineLength, lineHeight, line) {
-    var shiftX = (justify - horizontalAlign) * maxLineLength;
+function align(positionedGlyphs, justify, horizontalAlign, verticalAlign, maxLineLength, lineHeight, line, translate) {
+    var shiftX = (justify - horizontalAlign) * maxLineLength + translate[0];
```
Comment (lucaswoj) [inline js/symbol/shaping.js]: What about vertical translation?

File: debug/site.js [hunk 2]
```diff
@@ -54,6 +55,8 @@ map.on('style.load', function() {
         }
     });
 
+    map.setLayoutProperty('place-suburb', 'text-offset', [20,0]);
+
```
Comment (lucaswoj) [inline debug/site.js]: Please revert these changes

**Comment by kelvinabrokwa** [comment]
@samanpwbb @nickidlugash @ajashton @andreasviglakis @lucaswoj @ansis this PR fixes the multi-line text offset problem. However, it also uncovered another bug that has to do with offsetting center-justified text. Before this PR, they were only offset 1/2 of what the user entered and now they will be offset the correct amount. I'm not sure how much this will screw up styles but I wanted your input before merging.

**Comment by ansis** [comment]
> Before this PR, they were only offset 1/2 of what the user entered and now they will be offset the correct amount. I'm not sure how much this will screw up styles but I wanted your input before merging.

The good news is that most of the common cases are less affected. Text centered below/above a point isn't affected because it has no x-offset. Single-line labels that are anchor-left-and-justify-left or anchor-right-and-justify-right should also be unaffected.

I think the actual style breakage should be minimal

File: js/symbol/shaping.js (hunk 1 above)
Comment (ansis) [inline js/symbol/shaping.js]: vertical translation is still handled on line 36. It might be clearer to move it down here though

Comment (kelvinabrokwa) [inline js/symbol/shaping.js]: okay I'll move it down into `align` and then squash before merging

**Comment by lucaswoj** [comment]
:non-alcoholic-beer: Nice work @kelvinabrokwa! Its great having you on team GL JS!
